packages/nx/src/config/workspaces.ts

```diff
@@ -1,109 +1,14 @@
-import { dirname, join } from 'path';
-import { workspaceRoot } from '../utils/workspace-root';
-import { readJsonFile } from '../utils/fileutils';
-import { loadNxPlugins, loadNxPluginsSync } from '../utils/nx-plugin';
-
-import type { NxJsonConfiguration } from './nx-json';
-import { readNxJson } from './nx-json';
-import {
-  ProjectConfiguration,
-  ProjectsConfigurations,
-} from './workspace-json-project-json';
-import {
-  mergeAngularJsonAndProjects,
-  shouldMergeAngularProjects,
-} from '../adapter/angular-json';
-import { retrieveProjectConfigurationPaths } from '../project-graph/utils/retrieve-workspace-files';
-import {
-  buildProjectsConfigurationsFromProjectPathsAndPlugins,
-  mergeTargetConfigurations,
-  readTargetDefaultsForTarget,
-} from '../project-graph/utils/project-configuration-utils';
+import { dirname } from 'path';
 
+// TODO(v18): remove this class
+/**
+ * @deprecated These will be removed in v18.
+ * - For Workspaces.readWorkspaceConfiguration, use retrieveProjectConfigurations/retrieveProjectConfigurationsWithAngularProjects instead.
+ * - For Workspaces.readProjectsConfigurations, use retrieveProjectConfigurations/retrieveProjectConfigurationsWithAngularProjects instead.
```

Comment:
We removed this because we saw no one was using them. I don't think we need to give alternatives.

Reply:
also removed